[WIP] Documentation fixes and enhancements

harshbafna opened the pull request:
## Description

Fixes #561 

Skipped following doc fixes to avoid conflicts with already open PR :
 - refactoring batch inferencing documentation: done as part of #432 
 - refactoring custom service documentation: done as part of #432 
 - Adding README for vgg11: done as part of #559 

## Type of change

Please delete options that are not relevant.

- [x] Documentation fixes and enhancements

## Feature/Issue validation/testing

NA

## Checklist:
- [x] Have you made corresponding changes to the documentation?

harshbafna commented:
@maaquib : The following two codebuild log links for regression suite are broken in [test/README.md](https://github.com/pytorch/serve/blob/master/test/README.md) are broken. Could you please have a look into it :


https://torchserve-regression-test-gpu.s3.amazonaws.com/public/latest/test_exec.log 
https://torchserve-regression-test-gpu.s3.amazonaws.com/public/TS-NIGHTLY-REGRESSION-2020-07-10/tmp/test_exec.log

maaquib commented on examples/image_classifier/squeezenet/README.md:
````diff
@@ -0,0 +1,48 @@
+sm.save("squeezenet1_1.pt")
+```
+
+  * Save model using tracing
````
Inconsistent list level

shivamshriwas replied:
Corrected list level

maaquib commented on ci/buildspec.yml:
```diff
@@ -8,6 +8,10 @@ phases:
       - apt-get update
       - apt-get install -y curl gnupg2
       - pip install pip -U
+      - apt-get -y install nodejs-dev node-gyp libssl1.0-dev
```
Why the older version of libssl?

shivamshriwas replied:
removed libssl

maaquib commented on docker/README.md:
```diff
@@ -16,10 +16,10 @@
 
 ## First things first
 
+If you have not clone TorchServe source then:
```
NIT:
```suggestion
If you have not cloned TorchServe source then:
```

maaquib commented on docs/code_coverage.md:
```diff
@@ -1,5 +1,7 @@
 ## To check branch stability run the sanity suite as follows
 
+Ensure that node dependency are already installed in your system. Refer [Install Node dependency](#install-markdown-link-checker-dependencies)  
```
NIT:
```suggestion
Ensure that node dependency is already installed on your system. Refer [Install Node dependency](#install-markdown-link-checker-dependencies)  
```

maaquib commented on examples/image_classifier/squeezenet/README.md:
```diff
@@ -0,0 +1,48 @@
+#### Sample commands to create a squeezenet eager mode model archive, register it on TorchServe and run image prediction
+
+Run the commands given in following steps from the parent directory of the root of the repository. For example, if you cloned the repository into /home/my_path/serve, run the steps from /home/my_path
```
This instruction doesn't look correct. Example below should be `--model-file serve/examples/image_classifier/squeezenet/model.py` instead of `--model-file examples/image_classifier/squeezenet/model.py`

harshbafna replied:
Fixed the instructions.

maaquib commented on examples/image_classifier/squeezenet/README.md:
```diff
@@ -0,0 +1,48 @@
+
+#### TorchScript example using alexnet image classifier:
+
+* Save the Squeezenet1_1 model in as an executable script module or a traced script:
```
```suggestion
* Save the Squeezenet1_1 model as an executable script module or a traced script:
```

maaquib commented:
@harshbafna Please reasolve the merge conflicts on this

harshbafna commented:
@maaquib @dhanainme : 

I have resolved the conflicts and addressed the review comments.

The following links related to regression suite logs are still broken. Could you please look into this:

```
[✖] https://torchserve-regression-test-win.s3.amazonaws.com/public/latest/test_exec.log → Status: 403
[✖] https://torchserve-regression-test-gpu.s3.amazonaws.com/public/TS-NIGHTLY-REGRESSION-2020-10-06/tmp/test_exec.log → Status: 403
[✖] https://torchserve-regression-test-win.s3.amazonaws.com/public/TS-NIGHTLY-REGRESSION-2020-10-06/tmp/test_exec.log → Status: 403
```

maaquib commented:
@abishekchiffon @dhanainme Also the follwing seems to be broken

```
FILE: kubernetes/kfserving/image_transformer/README.md
[âœ–] ../../pytorch/README.md
[âœ“] https://github.com/kubeflow/kfserving/#install-kfserving
[âœ“] https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/
```

maaquib commented:
@harshbafna ping for merge conflicts
@dhanainme ping for broken batch links

lokeshgupta1975 commented:
Waiting for Dhana's changes for links, this will be merged after that.
Harsh to resolve the conflicts, so that this PR can be merged.

lokeshgupta1975 commented:
Once the builds have completed, Dhana should update the links in tests/README.md file and then this should pass.

maaquib commented:
@harshbafna Please resolve the merge conflict

dhanainme commented:
@harshbafna  / @dhaniram-kshirsagar . The Regression test results link is fixed. This seems to be failing for other links now.

harshbafna commented:
@dhanainme: Broken links post recent PR merges have been fixed.
